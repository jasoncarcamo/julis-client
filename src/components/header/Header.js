import React from 'react';
import TokenService from '../../services/TokenService';
import {Link} from 'react-router-dom';
import UserService from '../../services/UserService';
import './header.css'


export default class Header extends React.Component{
    constructor(props){
        super(props);
        this.state = {
            screenWidth: window.innerWidth
        }
    }

    renderLogIn = ()=>{
        return (
            <ul id="header-links">
                <Link to="/register" className="Link" onClick={this.closeMenu}>Sign Up</Link>
                <Link to="/login" className="Link" onClick={this.closeMenu}> Log In</Link>
            </ul>
        )
    }

    componentDidMount(){        
        window.addEventListener("resize", this.handleResize)
    }

    componentWillUnmount(){
        window.removeEventListener("resize", this.handleResize)
    }

    renderLogOut = ()=>{
        return (
            <ul id="header-links">
                <li><Link to="/user/services" className="Link" onClick={this.closeMenu}>services</Link></li>

                <li><Link to="/" onClick={this.handleLogOut} className="Link">Log Out</Link></li>
            </ul>
        )
    }

    handleLogOut = ()=>{
        UserService.clearId();
        TokenService.clearAuthToken();
        this.props.history.push('/')
        this.closeMenu()
    }

    handleMenuIcon = () => {
        const headerLinks = document.getElementById('header-links');
        const datePicker = document.getElementsByClassName('react-date-picker');
        const timePicker = document.getElementsByClassName('react-time-picker');

        if(headerLinks.style.display !== "flex"){

            headerLinks.style.display = "flex";

            if(datePicker[0] && timePicker[0]){                
            datePicker[0].style.display = "none";
            timePicker[0].style.display = "none";
            } else{
                return;
            }
        } else{            

            headerLinks.style.display = "none";

            if(datePicker[0] && timePicker[0]){                
                datePicker[0].style.display = "block";
                timePicker[0].style.display = "block";
            } else {
                return;
            }
        };    
    }

    handleResize=()=>{
        const headerLinks = document.getElementById('header-links')
        if(window.innerWidth > 445){
            headerLinks.style.display = "flex";
        } else{
            headerLinks.style.display = "none";
        }
        this.setState({screenWidth: window.innerWidth});
    }

    closeMenu = () =>{
        const headerLinks = document.getElementById("header-links");   
        if(headerLinks.style.display === "flex"){
            headerLinks.style.display = "none";
        }
    }

   
    render(){
        return (
            <header>
                <nav id="header-nav">
                    <div ><Link to="/" id="logo-icon">Julis Cleaning Service Inc.</Link></div>

                    <div id="menu-icon" onClick={this.handleMenuIcon}>
                        <div></div>
                        <div></div>
                        <div></div>
                    </div>

                    {TokenService.hasAuthToken() ? this.renderLogOut() : this.renderLogIn()}
                </nav>
            </header>
        )
    }
}