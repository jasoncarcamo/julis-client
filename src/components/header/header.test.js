import React from "react";
import ReactDOM from "react-dom";
import Header from "./Header";
import {BrowserRouter} from "react-router-dom";
import Enzyme from "enzyme";
import renderer from "react-test-renderer";
import Adapter from "enzyme-adapter-react-16";

Enzyme.configure({ adapter: new Adapter()});


describe("<Header/>", ()=>{
    it("Renders header ncompeonent", ()=>{
        const div = document.createElement("div");

        ReactDOM.render(<BrowserRouter><Header/></BrowserRouter>, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it("Create snapshot for header component", ()=>{
        const tree = renderer.create(<BrowserRouter><Header/></BrowserRouter>).toJSON();

        expect(tree).toMatchSnapshot();
    })

})