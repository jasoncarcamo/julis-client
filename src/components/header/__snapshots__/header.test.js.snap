// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> Create snapshot for header component 1`] = `
<header>
  <nav
    id="header-nav"
  >
    <div>
      <a
        aria-current="page"
        className="active"
        href="/"
        id="logo-icon"
        onClick={[Function]}
        style={Object {}}
      >
        Julis Cleaning Service Inc.
      </a>
    </div>
    <div
      id="menu-icon"
      onClick={[Function]}
    >
      <div />
      <div />
      <div />
    </div>
    <ul
      id="header-links"
    >
      <li>
        <a
          className="Link"
          href="/register"
          onClick={[Function]}
        >
          Sign Up
        </a>
      </li>
      <li>
        <a
          className="Link"
          href="/login"
          onClick={[Function]}
        >
           Log In
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
